src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -151,14 +153,24 @@ RelayEventExtendNames(Node *parseTree, char *schemaName, uint64 shardId)
 				{
 					Constraint *constraint = (Constraint *) command->def;
 					char **constraintName = &(constraint->conname);
+					const bool missingOk = false;
+					relationId = RangeVarGetRelid(alterTableStmt->relation,
+												  AccessShareLock,
+												  missingOk);
 
 					if (constraint->contype == CONSTR_PRIMARY && constraint->indexname)
 					{
 						char **indexName = &(constraint->indexname);
 						AppendShardIdToName(indexName, shardId);
 					}
 
-					AppendShardIdToName(constraintName, shardId);
+					/* append shardId to constraint names when necessary */
+					if (!PartitionedTable(relationId) ||
```

Comment:
My high level reaction here is that we should probably exercise all operations (such as`DROP/RENAME`) on the constraints with the test as much as possible, with whatever PG supports. 

Even if they are not allowed today on Citus, it might make sense to leave those tests in the regression test files -- with nice comments-- so that when implemented, we don't forget about this edge case.

Reply:
0049075 added ALTER and RENAME constraint tests that are expected to give meaningful errors.

We already have a DROP in the tests.

All `ALTER TABLE` commands that do not have `ADD CONSTRAINT` seems to be the of the following form:

```
ALTER TABLE [ IF EXISTS ] [ ONLY ] name [ * ]
    RENAME CONSTRAINT constraint_name TO new_constraint_name

ALTER TABLE [ IF EXISTS ] [ ONLY ] name [ * ]
    action [, ... ]

where action is one of:
...
    ALTER CONSTRAINT constraint_name [ DEFERRABLE | NOT DEFERRABLE ] [ INITIALLY DEFERRED | INITIALLY IMMEDIATE ]
    VALIDATE CONSTRAINT constraint_name
    DROP CONSTRAINT [ IF EXISTS ]  constraint_name [ RESTRICT | CASCADE ]
...
```

The only missing test case seems to be VALIDATE CONSTRAINT that appears to work just fine.

@onderkalaci Is it ok as it is?